packages/terra-date-picker/src/DateInput.jsx

```diff
@@ -115,6 +121,54 @@ class DatePickerInput extends React.Component {
     }
   }
 
+  static getDerivedStateFromProps(props, state) {
+    // Formats the selected date into a consistent value and sets input state.
+    const format = DateUtil.getDateFormat(props.intl.formatMessage({ id: 'Terra.datePicker.dateFormatOrder' }));
+    if (props.value !== state.prevDate) {
+      const tempDate = DateUtil.convertToISO8601(props.value, format);
+      return {
+        day: tempDate.substring(8, 10),
+        month: tempDate.substring(5, 7),
+        year: tempDate.substring(0, 4),
+        prevDate: props.value,
+      };
+    }
+    return null;
+  }
+
+  /**
+   * Moves focus to the correct input depending on date ordering.
+   * @param {string} inputValue - The value from the current input.
+   * @param {int} type - The input type, based on DateUtil.inputType.
+   * @param {string} dateVariant - Date variant based on the currently supported date
+   *                      orders: 'MM-DD-YYYY', 'DD-MM-YYYY', or 'YYYY-MM-DD'.
+   */
+  moveFocusOnChange(inputValue, type, dateVariant) {
+    if (dateVariant === 'MM-DD-YYYY') {
+      if (inputValue.length === 2) {
+        if (type === DateUtil.inputType.MONTH) {
+          this.dayInput.focus();
+        } else {
+          this.yearInput.focus();
+        }
+      }
+    } else if (dateVariant === 'DD-MM-YYYY') {
+      if (inputValue.length === 2) {
+        if (type === DateUtil.inputType.MONTH) {
+          this.yearInput.focus();
+        } else if (type === DateUtil.inputType.DAY) {
```

Comment:
Is there a reason why this needs the check for `type === DateUtil.inputType.DAY` but line 151 above doesn't?

Reply:
If you're in the month input, it needs to move to the day input. If you're in the year input, it doesn't matter. The only other option is that you're in the day input (for that date variant). The variant above could be rewritten the same way, since it is just swapping day and month.